submissions/5mountains/dom-api-task/js/main.js

```diff
@@ -0,0 +1,118 @@
+async function fetchAsync () {
+    let response = await fetch('./data/data.json');
+    let data = await response.json();
+    return data;
+}
+  
+fetchAsync()
+    .then(data => {
+        let details = createHtml(data);
+        drawPage(details);
+        toggleTabs();
+        animation();
+    })
+    .catch(reason => console.log(reason.message));
+
+function createHtml(data) {
+    const menuItems = document.createDocumentFragment();
+    const heroItems = document.createDocumentFragment();
+    data.pancakes.forEach((elem, index) => {
```

Comment:
This function looks WAAAAAY to big :D

Pls, split it into smaller parts. For this, you will need to move parts of code into separate functions, like `createImageWrapper`, etc. and call these functions in your callback.

BTW, instead of calling DOM API you can use template strings and write something like:

```javascript
const htmlContent = `
  <figure src={${src}}
    <a href=${href} /> 
  </figure>`;
```

etc. In such a way it will be more readable and understandable what DOM structure you are building.

Reply:
I need some time to do this (:

Reply:
Still looks messy. Please, separate into functions :)

Reply:
I feel that this time is gonna be OK. I split up maximum (in my point of view) into functions which, in their turn, call to other functions (:

Reply:
Yeah. Looks better now. You can resolve the conversation